src/docfx/lib/PathUtil.cs

```diff
@@ -0,0 +1,127 @@
+        /// <returns>The relative path</returns>
+        public static string GetRelativePathToFile(string fileRelativeTo, string path)
+        {
+            var directory = Path.GetDirectoryName(fileRelativeTo);
+            if (string.IsNullOrEmpty(directory))
+            {
+                return path;
+            }
+
+            return Path.GetRelativePath(directory, path);
+        }
+
+        /// <summary>
+        /// A normalized folder always ends with `/`, does not contain `\` and does not have consegative `.` or `/`.
+        /// </summary>
+        /// <param name="path">The folder path want to be normalized</param>
+        /// <returns>The normalized folder path</returns>
+        public static string NormalizeFolder(string path)
+        {
+            Debug.Assert(path.IndexOfAny(Path.GetInvalidPathChars()) < 0);
+
+            var str = Normalize(path);
+            if (str.Length == 0 || str == "/")
+            {
+                return "./";
+            }
+
+            if (!str.EndsWith('/'))
+            {
+                str += '/';
+            }
+            return str;
+        }
+
+        /// <summary>
+        /// A normalized file cannot end with `/`, does not contain `\` and does not have consegative `.` or `/`.
+        /// </summary>
+        /// <param name="path">The file path want to be normalized</param>
+        /// <returns>The normalized file path</returns>
+        public static string NormalizeFile(string path)
+        {
+            Debug.Assert(Path.GetDirectoryName(path).IndexOfAny(Path.GetInvalidPathChars()) < 0);
+            Debug.Assert(Path.GetFileName(path).IndexOfAny(Path.GetInvalidFileNameChars()) < 0);
+
+            return Normalize(path);
+        }
+
+        private static string Normalize(string path)
+        {
+            path = path.Replace('\\', '/');
+
+            var needReorder = false;
+            if (path.IndexOf('.') != -1)
+            {
+                needReorder = true;
+            }
+
+            if (!needReorder)
+            {
+                return path;
+            }
+
+            var parentCount = 0;
+            var rooted = path[0] == '/';
+            var stack = new List<string>();
+            foreach (var segment in path.Split('/'))
+            {
+                if (segment == ".." && stack.Count > 0)
+                    stack.RemoveAt(stack.Count - 1);
+                else if (segment == "..")
+                    parentCount++;
+                else if (segment != "." && !string.IsNullOrEmpty(segment))
+                    stack.Add(segment);
+            }
+
+            var res = new StringBuilder();
+            if (rooted)
+            {
+                res.Append('/');
+            }
+            else
+            {
+                while (parentCount-- > 0)
+                {
+                    res.Append("../");
+                }
+            }
+
+            var i = 0;
```

Comment:
>i [](start = 16, length = 1)

use bool? seems only care about whether it's 0.

Reply:
i believe both would be ok

---
In reply to: [180007222](https://github.com/dotnet/docfx/pull/2601#discussion_r180007222) [](ancestors = 180007222)